hub/core/dataset/dataset.py

```diff
@@ -1488,6 +1489,8 @@ def rename(self, path: Union[str, pathlib.Path]):
         Raises:
             RenameError: If `path` points to a different directory.
         """
+        self.path = convert_windows_path_to_linux_if_needed(self.path)
```

Comment:
shouldn't we be converting the incoming path and not the path that's already present in the dataset?
what if the input to rename is a windows path with \\, rename would still fail right?